File: 1.0/src/cz/mg/vulkan/PFNvkCmdPushConstants.java
package cz.mg.vulkan;

public class PFNvkCmdPushConstants extends VkFunctionPointer {
    public PFNvkCmdPushConstants() {
    }

    public PFNvkCmdPushConstants(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCmdPushConstants(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdPushConstants(long value) {
        setValue(value);
    }

    public PFNvkCmdPushConstants(VkInstance instance) {
        super(instance, new VkString("vkCmdPushConstants"));
    }

    public void call(VkCommandBuffer commandBuffer, VkPipelineLayout layout, VkShaderStageFlags stageFlags, VkUInt32 offset, VkUInt32 size, VkObject pValues){
        call(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddress(), layout != null ? layout.getVkAddress() : VkPointer.getNullAddress(), stageFlags != null ? stageFlags.getVkAddress() : VkPointer.getNullAddress(), offset != null ? offset.getVkAddress() : VkPointer.getNullAddress(), size != null ? size.getVkAddress() : VkPointer.getNullAddress(), pValues != null ? pValues.getVkAddress() : VkPointer.NULL);
    }

    protected static native void call(long vkaddress, long commandBuffer, long layout, long stageFlags, long offset, long size, long pValues);
}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkCmdPushConstants.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdPushConstants_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong layout, jlong stageFlags, jlong offset, jlong size, jlong pValues){
    (void)env;
    (void)jc;
    PFN_vkCmdPushConstants f = (PFN_vkCmdPushConstants)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkPipelineLayout*)jniLongToPointer(layout)),
        *((VkShaderStageFlags*)jniLongToPointer(stageFlags)),
        *((uint32_t*)jniLongToPointer(offset)),
        *((uint32_t*)jniLongToPointer(size)),
        ((void*)jniLongToPointer(pValues))
    );
}


void Java_cz_mg_vulkan_PFNvkCmdPushConstants_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong layout, jint stageFlags, jint offset, jint size, jlong pValues){
    (void)env;
    (void)jc;
    PFN_vkCmdPushConstants f = (PFN_vkCmdPushConstants)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkPipelineLayout*)jniLongToPointer(layout)),
        (VkShaderStageFlags)stageFlags,
        (uint32_t)offset,
        (uint32_t)size,
        ((void*)jniLongToPointer(pValues))
    );
}

